src/Functions/FunctionsLanguageClassification.cpp

```diff
@@ -0,0 +1,126 @@
+#if !defined(ARCADIA_BUILD)
+#    include "config_functions.h"
+#endif
+
+#if USE_CLD2
+
+#include <Functions/FunctionsTextClassification.h>
+#include <Functions/FunctionFactory.h>
+
+#include "compact_lang_det.h"
+
+namespace DB
+{
+/* Determine language of Unicode UTF-8 text.
+ * Uses the cld2 library https://github.com/CLD2Owners/cld2
+ */
+template <bool mixed>
+struct LanguageClassificationImpl
+{
+
+    using ResultType = String;
+
+
+    static void constant(String data, String & res)
+    {
+        bool is_UTF8 = true;
+        const char * str = data.c_str();
+        if (!mixed)
+        {
+            String ans(LanguageName(CLD2::DetectLanguage(str, strlen(str), true, &is_UTF8)));
+            res = ans;
+        }
+        else
+        {
+            CLD2::Language result_lang_top3[3];
+            int pc[3];
+            int bytes[3];
+            CLD2::DetectLanguageSummary(str, strlen(str), true, result_lang_top3, pc, bytes, &is_UTF8);
```

Comment:
Check for ISO validness?

Being ISO-639-1 I was using this code. Sometimes it returns Latn or simplified Chinese languages

```cpp
  absl::ConsumeSuffix(&code_string, "-Latn");
  absl::ConsumeSuffix(&code_string, "-Hant");
  if (code_string == "iw") {
    return "he";
  }
  if (code_string == "jw") {
    return "jv";
  }
  if (code_string == "in") {
    return "id";
  }
  if (code_string == "mo") {
    return "ro";
  }
  if (code_string.size() != 2) {
    return "other";
```

I won the contest for detecting languages of small texts by using CLD2+CLD3. Pseudocode

```cpp
  const auto lang = CLD2::DetectLanguage(combined.data(), combined.size(), true,
                                         &is_reliable);
  std::string code_string = AdjustCodeString(LanguageCode(lang));
  if (!is_reliable) {
    auto [another_code_string, cld3_is_reliable] = CLD3(combined);
    if (cld3_is_reliable &&
        combined.size() > static_cast<size_t>(resources::Config()->min_unreliable_cld3_bytes())) {
      code_string = AdjustCodeString(another_code_string);
      is_reliable = cld3_is_reliable;
    }
  }
```

CLD3 does not rely on chromium code